lib/puppet/file_serving/http_metadata.rb

```diff
@@ -20,7 +20,21 @@ def initialize(http_response, path = '/dev/null')
       checksum = checksum.unpack("m0").first.unpack("H*").first
       @checksums[:md5] = "{md5}#{checksum}"
     end
+    # Add support for artifactory headers.
+    # This adds support for all three algorithms, sha256, sha1, and md5.
+    if checksum = http_response['X-Checksum-Md5']
```

Comment:
Does artifactory require this specific header format? I'd be slightly more warm and fuzzy if we didn't have to insert a different header for every checksum type we support.

Also, I'm curious; what exactly are you doing with artifactory interfacing with the Puppet http file metadata endpoint?